taglib/fileref.cpp

```diff
@@ -54,6 +54,62 @@
 
 using namespace TagLib;
 
+template <typename T>
+static File* createInternal(T arg, const String& ext, bool readAudioProperties,
```

Comment:
The extension extraction code could be here as well. Just pass it `FileName fileName` instead of `String ext`.